### PR TITLE
feat: use workpool to reduce operator memory usage

### PR DESCRIPTION
**Pull request by 0xff-dev**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it
feat: use workpool to reduce operator memory usage

#### Which issue(s) this PR fixes

Fixes #

### CODE REVIEW

File: controllers/component_controller.go [hunk 1]
```diff
@@ -195,27 +194,7 @@ func (r *ComponentReconciler) UpdateValuesConfigmap(ctx context.Context, logger
 		versionStr := version.Version // https://golang.org/doc/faq#closures_and_goroutines
 		httpDonwloadURLs := version.URLs
 		g.Go(func() error {
```
Comment (0xff-dev) [inline controllers/component_controller.go:196]: Should we not wait for the goroutine to complete?  The action completed by goroutine is to push the structure into chan. @bjwswang @Abirdcfly

File: pkg/helm/helm_chart_workpool.go [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+	defer os.Remove(dir)
+
+	b, err := os.ReadFile(dir + "/" + entryName + "/values.yaml")
+	if err != nil {
+		c.logger.Error(err, "")
+		return err
+	}
+
+	cm.Data[v1alpha1.ValuesConfigMapKey] = string(b)
+	b, err = os.ReadFile(dir + "/" + entryName + "/README.md")
```
Comment (0xff-dev) [inline pkg/helm/helm_chart_workpool.go:194]: Should an error be reported when the readme does not exist?

**Comment by bjwswang** [comment]
@0xff-dev Revert the default manager memory, then run the tests.

File: controllers/component_controller.go (hunk 1 above)
Comment (Abirdcfly) [inline controllers/component_controller.go:196]: there is `g.SetLimit(workers)` in line 192, IMO, we should set this limit to `helm.ChartWorker` and no need to use `g.Go()`

**Review by Abirdcfly — DISMISSED** [review]

**Review by Abirdcfly — APPROVED** [review]

**Comment by 0xff-dev** [comment]
image
According to local tests, limiting memory growth by  `workpool`, the maximum memory used is `1.4G`. so adjust the deployment's memory limit value to `1.5G`.

**Comment by bjwswang** [comment]
> image
> According to local tests, limiting memory growth by `workpool`, the maximum memory used is `1.4G`. so adjust the deployment's memory limit value to `1.5G`.

Can you pls show your test details?  From my point of view, 1.5G still too much...

**Comment by bjwswang** [comment]
@0xff-dev We can use this for v0.2.0. And do further optimizations at v0.2.x maybe

**Review by bjwswang — APPROVED** [review]

**Pull request merged**